Sources/SwiftDocC/Model/DocumentationNode.swift

```diff
@@ -512,7 +547,36 @@ public struct DocumentationNode {
             let docCommentMarkup = Document(parsing: docCommentString, source: docCommentLocation?.url, options: documentOptions)
             let offset = symbol.docComment?.lines.first?.range
 
-            let docCommentDirectives = docCommentMarkup.children.compactMap({ $0 as? BlockDirective })
+            var docCommentDirectives = docCommentMarkup.children.compactMap { $0 as? BlockDirective }
+            
+            var problems = [Problem]()
+            
+            if let bundle, let context {
+                (metadata, docCommentDirectives) = DirectiveParser()
+                    .parseSingleDirective(
+                        Metadata.self,
+                        from: docCommentDirectives,
+                        parentType: Symbol.self,
+                        source: docCommentLocation?.url,
+                        bundle: bundle,
+                        context: context,
+                        problems: &problems
+                    ) as! (Metadata?, [BlockDirective])
```

Comment:
It was because the method returned `any Markup` and we needed to force cast the second element of the tuple so that it can be assigned to `docCommentDirectives`. But with your suggestion of making the `markupElements` argument `inout`, this is no longer needed.